include/crow/routing.h

```diff
@@ -972,6 +972,25 @@ namespace crow
             unsigned rule_index = found.first;
             if (!rule_index)
             {
+                for (auto& per_method: per_methods_)
+                {
+                    auto& trie = per_method.trie;
+                    auto& rules = per_method.rules;
+
+                    auto found = trie.find(req.url);
+
+                    unsigned rule_index = found.first;
```

Comment:
Maybe extract lines 977-982 into method? There are 4 same code blocks in this file ;)

Reply:
So these lines are repeated for `handle_upgrade()` and `handle()`, but some of the variables [`trie`, `rules`, `found`] are being used, so I condensed the parts that didn't require the variables, there's still a bit of repetition, but I don't think it's a big deal (it's also less repetition than the original code).

Reply:
Ok. Thanks for little cleanup ;)